File: mns-rest/src/main/java/cz/fb/manaus/rest/ExceptionCounter.java
package cz.fb.manaus.rest;

import com.codahale.metrics.MetricRegistry;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.ControllerAdvice;
import org.springframework.web.bind.annotation.ExceptionHandler;

@ControllerAdvice
public class ExceptionCounter {

    public static final String METRIC_NAME = "_ERROR_";
    @Autowired
    private MetricRegistry metricRegistry;

    @ExceptionHandler(value = Throwable.class)
    public void defaultErrorHandler(Throwable e) throws Throwable {
        metricRegistry.counter(METRIC_NAME).inc();
        throw e;
    }
}

File: mns-rest/src/main/java/cz/fb/manaus/rest/ExceptionCounter.kt
package cz.fb.manaus.rest

import io.micrometer.core.instrument.Metrics
import org.springframework.web.bind.annotation.ControllerAdvice
import org.springframework.web.bind.annotation.ExceptionHandler

@ControllerAdvice
class ExceptionCounter {

    @ExceptionHandler(value = [Throwable::class])
    @Throws(Throwable::class)
    fun defaultErrorHandler(e: Throwable) {
        Metrics.counter("exception_count").increment()
        throw e
    }

}